lib/IDE/Refactoring.cpp

```diff
@@ -3887,7 +3889,1442 @@ bool RefactoringActionConvertToComputedProperty::performChange() {
+        bool AddPlaceholder = Placeholders.count(D);
+        StringRef Name = newNameFor(D, false);
+        if (AddPlaceholder || !Name.empty())
+          return addCustom(DRE->getStartLoc(),
+                           Lexer::getLocForEndOfToken(SM, DRE->getEndLoc()),
+                           [&]() {
+                             if (AddPlaceholder)
+                               OS << PLACEHOLDER_START;
+                             if (!Name.empty())
+                               OS << Name;
+                             else
+                               D->getName().print(OS);
+                             if (AddPlaceholder)
+                               OS << PLACEHOLDER_END;
+                           });
+      }
+    } else if (auto *FTE = dyn_cast<ForceValueExpr>(E)) {
+      if (auto *D = FTE->getReferencedDecl().getDecl()) {
+        if (Unwraps.count(D))
+          return addCustom(FTE->getStartLoc(),
+                           FTE->getEndLoc().getAdvancedLoc(1),
+                           [&]() { OS << newNameFor(D, true); });
+      }
+    } else if (NestedExprCount == 1) {
+      if (CallExpr *CE = TopHandler.isCall(E))
+        return addCustom(CE->getStartLoc(), CE->getEndLoc().getAdvancedLoc(1),
+                         [&]() { addHandlerCall(CE); });
+
+      auto *CE = dyn_cast_or_null<CallExpr>(E);
+      auto HandlerDesc = AsyncHandlerDesc::find(CE, /*ignoreName=*/true);
+      if (HandlerDesc.isValid())
+        return addCustom(CE->getStartLoc(), CE->getEndLoc().getAdvancedLoc(1),
+                         [&]() { addAsyncAlternativeCall(CE, HandlerDesc); });
+    }
+
+    return true;
+  }
+#undef PLACEHOLDER_START
+#undef PLACEHOLDER_END
+
+  bool walkToExprPost(Expr *E) override {
+    NestedExprCount--;
+    return true;
+  }
+
+  bool addCustom(SourceLoc End, SourceLoc NextAddedLoc,
+                 std::function<void()> Custom = {}) {
+    addRange(LastAddedLoc, End);
+    Custom();
+    LastAddedLoc = NextAddedLoc;
+    NestedExprCount--;
+    return false;
+  }
+
+  void addRange(SourceLoc Start, SourceLoc End, bool ToEndOfToken = false) {
+    if (ToEndOfToken) {
+      OS << Lexer::getCharSourceRangeFromSourceRange(SM,
+                                                     SourceRange(Start, End))
+                .str();
+    } else {
+      OS << CharSourceRange(SM, Start, End).str();
+    }
+  }
+
+  void addRange(SourceRange Range, bool ToEndOfToken = false) {
+    addRange(Range.Start, Range.End, ToEndOfToken);
+  }
+
+  void addFuncDecl(const FuncDecl *FD) {
+    assert(FD->hasBody() && "Must have a body");
+
+    auto *Params = FD->getParameters();
+
+    // First chunk: start -> the parameter to remove (if any)
+    SourceLoc LeftEndLoc = Params->getLParenLoc().getAdvancedLoc(1);
+    if (TopHandler.Index - 1 >= 0) {
+      LeftEndLoc = Lexer::getLocForEndOfToken(
+          SM, Params->get(TopHandler.Index - 1)->getEndLoc());
+    }
+    addRange(FD->getStartLoc(), LeftEndLoc);
```

Comment:
I think StartLoc points to 'static' if there is one and 'func' otherwise. Should this include other leading attributes/modifiers too?

Reply:
Thanks, nice catch - will use the range with attributes instead.